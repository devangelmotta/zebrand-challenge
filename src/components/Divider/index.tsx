import React, { memo } from 'react';
import styled from 'styled-components';

const Divider = styled.div`
  display: ${(props: Props) => props.display || "flex"};
  position: ${props => props.position || "relative"};
  width: ${props => props.width || "1px"};
  max-width: ${props => props.maxWidth || "none"};
  min-width: ${props => props.minWidth || "none"};
  height: ${props => props.height || "1px"};
  max-height: ${props => props.maxHeight || "none"};
  background-color: ${props => props.color || "#ddd"};
  padding: ${props => props.padding || "0 0 0 0"};
  border: ${props => props.border || "0 solid transparent"};
  border-radius: ${props => props.borderRadius || "0"};
  margin: ${props => props.margin || "0 0 0 0"};
  z-index: ${props => props.zIndex || "auto"};
  -webkit-box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  -moz-box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  box-shadow: ${props => props.shadow || "0px 0px 0px 0px rgba(0,0,0,0)"}; 
  `;

interface Props {
  readonly display? : string,
  readonly flexDirection? : string,
  readonly justifyContent? : string,
  readonly position? : string,
  readonly alignContent? : string,
  readonly alignItems? : string,
  readonly width? : string,
  readonly height? : string,
  readonly maxHeight? : string,
  readonly minWidth? : string,
  readonly maxWidth? : string,
  readonly backgroundColor? : string,
  readonly background? : string,
  readonly padding? : string,
  readonly border? : string,
  readonly borderRadius? : string,
  readonly margin? : string,
  readonly hoverBgColor? : string,
  readonly hoverBorder? : string,
  readonly zIndex?: string
  readonly shadow?: string;
}

export default memo(Divider);

