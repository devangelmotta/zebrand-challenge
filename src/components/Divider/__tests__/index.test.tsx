import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import Divider from '..';

test('it works', () => {
  const tree = renderer.create(<Divider />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('display', 'flex')
  expect(tree).toHaveStyleRule('position', 'relative')
  expect(tree).toHaveStyleRule('width', '1px')
  expect(tree).toHaveStyleRule('max-width', 'none')
  expect(tree).toHaveStyleRule('min-width', 'none')
  expect(tree).toHaveStyleRule('height', '1px')
  expect(tree).toHaveStyleRule('max-height', 'none')
  expect(tree).toHaveStyleRule('background-color', '#404040')
  expect(tree).toHaveStyleRule('padding', '0 0 0 0')
  expect(tree).toHaveStyleRule('border', '0 solid transparent')
  expect(tree).toHaveStyleRule('border-radius', '0')
  expect(tree).toHaveStyleRule('margin', '0 0 0 0')
  expect(tree).toHaveStyleRule('z-index', 'auto')
  expect(tree).toHaveStyleRule('-webkit-box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
  expect(tree).toHaveStyleRule('-moz-box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
  expect(tree).toHaveStyleRule('box-shadow', '0px 0px 0px 0px rgba(0,0,0,0)')
})
