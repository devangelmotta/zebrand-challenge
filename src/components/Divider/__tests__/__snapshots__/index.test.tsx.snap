// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 1px;
  max-width: none;
  min-width: none;
  height: 1px;
  max-height: none;
  background-color: transparent;
  padding: 0 0 0 0;
  border: 0 solid transparent;
  border-radius: 0;
  margin: 0 0 0 0;
  z-index: auto;
  -webkit-box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
  -moz-box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
  box-shadow: 0px 0px 0px 0px rgba(0,0,0,0);
}

<div
  className="c0"
/>
`;
